rp2040-hal/src/uart/utils.rs

```diff
@@ -72,7 +73,7 @@ pub enum Parity {
 #[non_exhaustive]
 pub struct UartConfig {
     /// The baudrate the uart will run at.
-    pub baudrate: HertzU32,
+    pub baudrate: u32,
```

Comment:
If the type does not carry the unit, I think it should be at least documented.

```suggestion
    /// The baudrate in bit per second the uart will run at.
    pub baudrate: u32,
```

Reply:
The unit of a baudrate is a baud, not bit per second. So it's already in the name.
https://en.wikipedia.org/wiki/Baudrate

For an uart which only transfers one bit per symbol, the numeric values are the same, so the units are often confused. But I'd rather not increase that confusion by documenting the unit of a variable named "baudrate" with a unit of "bit per second".

Reply:
Yes, baud is symbols per second. For uart a symbol is a bit so baud is equal to bit per second.
Also baudrate is already redundant in itself (since baud is already a rate) but that's commonly used in the industry 🤷 .